### PR TITLE
catalog fix

### PR DESCRIPTION
**Pull request by StrongMonkey**

### CODE REVIEW

File: api/setup/setup.go [hunk 1]
```diff
@@ -31,6 +31,7 @@ func Schemas(ctx context.Context, management *config.ManagementContext, schemas
 	Templates(schemas)
 	ClusterRegistrationTokens(schemas)
 	addUserAction(schemas)
+	addCatalogRefreshAction(schemas)
```
Comment (ibuildthecloud) [inline api/setup/setup.go]: I'd prefer these two methods to be Users(schemas) and Catalogs(schemas) to match the style above

Comment (StrongMonkey) [reply api/setup/setup.go]: 👍

File: api/catalog/filter.go [hunk 2]
```diff
@@ -6,15 +6,57 @@ import (
...
 
 func TemplateFormatter(apiContext *types.APIContext, resource *types.RawResource) {
 	delete(resource.Values, "icon")
+	resource.Values["versionLinks"] = extractVersionLinks(apiContext, resource)
+	delete(resource.Values, "version")
 	resource.Links["icon"] = apiContext.URLBuilder.Link("icon", resource)
+	catalogName := strings.Split(resource.ID, "-")[0]
+	// idk why current returns a url start with http:/ instead of http://
+	// but this link is hacky
+	resource.Links["catalog"] = strings.Replace(path.Join(path.Dir(apiContext.URLBuilder.Current()), "catalogs", catalogName), "http:/", "http://", 1)
```
Comment (ibuildthecloud) [inline api/catalog/filter.go]: My eyes.... my poor eyes are bleeding.  Add a method to URLBuilder for `ResourceLinkByID(schema *Schema, id string)`

**Comment by StrongMonkey** [comment]
https://github.com/rancher/norman/pull/20
https://github.com/rancher/types/pull/72
https://github.com/rancher/catalog-controller/pull/1
related pr. Don't merge until dep PRs are merged.

**Pull request merged**